rfcs/01780_policy_delegation.md

```diff
@@ -0,0 +1,308 @@
+that means in practice (e.g. a version with bug fixes or improved
+functionality).
+
+### Wasm Module Signature
+
+https://github.com/project-oak/oak/blob/573e61e74668db3eb53747f38757b00aa899f1f6/oak_abi/proto/label.proto#L71-L77
+
+In order to facilitate more useful data flows with declassification other than
+just individual Wasm modules identified by a precise hash, Oak also supports
+Wasm module signature labels.
+
+A Wasm module signature principal refers to a public key that is used to sign
+any number of individual Wasm modules (or more precisely, to sign their hashes).
+
+For instance, let's assume there are three Wasm modules, with hashes
+respectively w_0, w_1, w_2. Also there are two (public, secret) key pairs (p_0,
+s_0), (p_1, s_1).
+
+Let's assume that the modules are supposed to implement some declassification
+logic, e.g. counting the number of people in an image given raw pixels. s_0 and
+s_1 are two parties that verify such models (from correctness, privacy and
+security point of view) and sign them if they (respectively) believe that they
+are trustworthy.
+
+The owner of s_0 signs w_0 and w_1 with s_0, while the owner of s_1 signs w_1
+and w_2 with s_1:
+
+```text
+    ┌ w_0
+s_0 │
+    └ w_1 ┐
+          │ s_1
+      w_2 ┘
+```
+
+## Delegation Manifest
+
+The main contribution of this RFC is to specify a mechanism by which parties may
+publish a list of hashes in a standard format and serve it statically over
+HTTPS.
+
+In the example above, the owner of s_0 would need to publish the signed module
+hashes somewhere, and a client would have to know the value of s_0 itself (the
+actual key used for signing the module), which introduces extra complexity and
+parties around the deployment and publishing of new modules.
+
+Here we introduce a delegation mechanism that is based on the publication of a
+manifest file on an HTTPS endpoint.
+
+A Delegation Manifest is identified (and located) by an HTTPS URL pointing to a
+JSON file with the schema described below.
+
+For instance, Google may publish a delegation manifest under
+`https://google.com/oak/manifests/count_people_from_pixels.json` with the
+following contents:
+
+```json
+{
+  "trusted": [
+    {
+      "type": "manifest",
+      "value": "https://eff.com/oak/manifests/count_people_from_pixels.json"
+    },
+    {
+      "type": "wasm_module",
+      "value": "65c296d60c077db03c7ec90c60107a52221c14bb"
+    },
+    {
+      "type": "wasm_module",
+      "value": "cf26b96746311908f8fe40340a8b6c1f06f5a1b5"
+    }
+  ]
+}
+```
+
+- ~~The `description` field contains a human-readable description of the entries
+  that the manifest refers to.~~ This was considered but then removed, so that
+  it may not be misused by showing it to users in the wrong context.
+- The `trusted` field contains a list of object that are trusted by the
+  publisher to conform to the abstract policy identified by the manifest. Each
```

Comment:
How does the manifest identify the abstract policy?

Reply:
It does not, the policy is expected to live on a web page or some other document, and just link to the manifest (now the public key -- PTAL)